src/riak_kv_backend.erl

```diff
@@ -231,62 +206,55 @@ fold_objects({Backend, State}) ->
+                                           [],
+                                           State),
+                     lists:sort(Keys)
+                 end),
 
-             ?assertEqual([{{B1,K1}, <<"v1">>}],
-                          begin
-                              {ok, Objects1} =
-                                  Backend:fold_objects(FoldObjectsFun,
-                                                       [],
-                                                       [],
-                                                       State),
-                              lists:sort(ObjFilter(Objects1))
-                          end),
-             ?assertMatch({ok, _},
-                          Backend:put(B3, K3, [], <<"v3">>, State)),
-             ?assertEqual([{{B1,K1},<<"v1">>},
-                           {{B3,K3},<<"v3">>}],
-                          begin
-                              {ok, Objects} =
-                                  Backend:fold_objects(FoldObjectsFun,
-                                                       [],
-                                                       [],
-                                                       State),
-                              lists:sort(ObjFilter(Objects))
-                          end)
-     end
-    }.
+    ?assertEqual([{{B1,K1}, <<"v1">>}],
+                 begin
+                     {ok, Objects1} =
+                         Backend:fold_objects(FoldObjectsFun,
+                                              [],
+                                              [],
+                                              State),
+                     lists:sort(ObjFilter(Objects1))
+                 end),
+    {ok, State2} =  Backend:put(B3, K3, [], <<"v3">>, State),
+    ?assertEqual([{{B1,K1},<<"v1">>},
+                  {{B3,K3},<<"v3">>}],
+                 begin
+                     {ok, Objects} =
+                         Backend:fold_objects(FoldObjectsFun,
+                                              [],
+                                              [],
+                                              State2),
+                     lists:sort(ObjFilter(Objects))
+                 end),
+    empty_check(Backend, State).
 
-empty_check({Backend, State}) ->
+empty_check(Backend, State) ->
     {B1, _B2, K1, _K2} = make_bs_and_ks(Backend),
     B3 = make_test_bucket(Backend, 3),
     K3 = make_test_key(Backend, 3),
-    {"is_empty test",
-     fun() ->
-             ?assertEqual(false, Backend:is_empty(State)),
-             ?assertMatch({ok, _}, Backend:delete(B1,K1, [], State)),
-             ?assertMatch({ok, _}, Backend:delete(B3,K3, [], State)),
-             ?assertEqual(true, Backend:is_empty(State))
-     end
-    }.
+    ?assertEqual(false, Backend:is_empty(State)),
+    {ok, State2} = Backend:delete(B1,K1, [], State),
+    {ok, State3} = Backend:delete(B3,K3, [], State2),
+    ?assertEqual(true, Backend:is_empty(State3)),
```

Comment:
Wow, so we rely on the B1 and B3 keys to already exist (and no other keys to exist) as inherited from the previous test. I guess that's OK, it's just a bit odd to see lines 250-252 doing gyrations to make the exact two keys that we're expecting to already be there.  I guess I'd expected (perhaps to insert some keys into a known empty state), then a fold to enumerate all keys, the delete them all.  {shrug}